src/uri/port.rs

```diff
@@ -51,14 +51,99 @@ impl<'a> Port<'a> {
     }
 }
 
+impl<'a> fmt::Display for Port<'a> {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(fmt, "{}", self.bytes)
+    }
+}
+
+impl<'a> AsRef<str> for Port<'a> {
+    fn as_ref(&self) -> &str {
+        self.as_str()
+    }
+}
+
+impl<'a> Into<u16> for Port<'a> {
+    fn into(self) -> u16 {
+        self.as_u16()
+    }
+}
+
+impl<'a> Into<&'a str> for Port<'a> {
+    fn into(self) -> &'a str {
+        self.as_str()
+    }
+}
+
+
 impl<'a> PartialEq for Port<'a> {
     fn eq(&self, other: &Self) -> bool {
         self.port == other.port
     }
 }
 
-impl<'a> fmt::Display for Port<'a> {
-    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
-        write!(fmt, "{}", self.bytes)
+macro_rules! port_partialeq {
+    ($t:ty) => {
+        impl<'a> PartialEq<$t> for Port<'a> {
+            fn eq(&self, port: &$t) -> bool {
+                self.port as $t == *port
+            }
+        }
+    }
+}
+
+port_partialeq!(u16);
+port_partialeq!(u32);
+port_partialeq!(u64);
+
+port_partialeq!(i16);
```

Comment:
At a glance, an unchecked cast from `u16` -> `i16` is problematic unless there is a guarantee that the value will not be greater than `i16::MAX`. Where is this checked?

Reply:
Added the check in